### DISCUSSION TITLE
TypeError caused by HasMetaDataFile

### DISCUSSION THREAD

**Discussion by mtiessen1175** [biigle/core#986] [Q&A] [answered]
After pulling the latest changes of the _dev-modules_ branch, I receive a TypeError when trying to open the Edit-page of a Volume.
The TypeError: `stream_copy_to_stream(): Argument #1 ($from) must be of type resource, null given ` 
Full description of the error in app/Traits/HasMetadataFile.php :49:
```
44 return Cache::store('array')->remember($key, 60, function () use ($disk) {
45    $tmpPath = tempnam(sys_get_temp_dir(), 'metadata');
46    try {
47        $to = fopen($tmpPath, 'w');
48        $from = Storage::disk($disk)->readStream($this->metadata_file_path);
49        stream_copy_to_stream($from, $to);
50        $type = ($this->media_type_id === MediaType::imageId()) ? 'image' : 'video';
51
52        $parser = new $this->metadata_parser(new SplFileInfo($tmpPath));
53
54        return $parser->getMetadata();
55 } finally {
56        if (isset($to) && is_resource($to)) {
57            fclose($to);
58        }
59        File::delete($tmpPath);
60 }
```
**Note:** This error only happens on volumes where I manually uploaded the metadata files (BIIGLE CSV in my example case).

**Comment by mzur** [comment]
You probably have to run the migrations with `docker compose exec -it -u0 worker php artisan migrate`.

**Reply by mtiessen1175** [reply]
@mzur, thanks for your answer!
Running the suggested command states "Nothing to migrate". Should I instead try to rollback the migrations, e.g. by using `migrate:refresh`?

**Reply by mzur** [reply, marked as answer]
Please open the console (`docker compose exec -it -u0 worker php artisan tinker`) and get the volume with `Volume::find(xxx)` (use the volume ID). Is `metadata_file_path` set? Do you find the metadata file at this location in the metadata storage disk?

**Reply by mtiessen1175** [reply]
The `metadata_file_path` is set for the volume ("4.csv"), but the file was indeed missing in the storage (or at least named differently; "2.csv"). Thanks for the guidance!
